packages/core/src/types/block-production.types.ts

```diff
@@ -4,7 +4,44 @@ import BigNumber = require('bn.js')
 /* Internal Imports */
 import { AbiStateUpdate } from '../app'
 
+export interface MerkleIntervalTreeNode {
+  hash: Buffer // Hash of the sibling or leaf data
+  start: Buffer // The start interval value for this node
+  data: Buffer // concatenation of (hash, index)
+}
+
+export interface MerkleIntervalInclusionProof {
+  siblings: MerkleIntervalTreeNode[]
+  leafPosition: BigNumber
+}
+
+export interface MerkleIntervalProofOutput {
+  root: MerkleIntervalTreeNode
+  maxEnd: BigNumber
+}
+
+export interface MerkleIntervalTree {
+  dataBlocks: any
+  levels: MerkleIntervalTreeNode[][]
+  root(): MerkleIntervalTreeNode
+  getInclusionProof(leafposition: number): MerkleIntervalInclusionProof
+}
+
 export interface SubtreeContents {
   assetId: Buffer
   stateUpdates: AbiStateUpdate[]
 }
+
+export interface DoubleMerkleIntervalTree extends MerkleIntervalTree {
+  dataBlocks: SubtreeContents
+  subtrees: MerkleIntervalTree[]
+  getStateUpdateInclusionProof(
+    stateUpdatePosition: number,
+    assetIdPosition: number
+  ): DoubleMerkleInclusionProof
+}
+
+export interface DoubleMerkleInclusionProof {
```

Comment:
Would be good to add a comment with a link to our spec http://spec.plasma.group/en/latest/src/01-core/double-layer-tree.html